app/src/main/java/org/schabi/newpipe/util/external_communication/TextLinkifier.java

```diff
@@ -0,0 +1,281 @@
+
+        while (hashtagsMatches.find()) {
+            final int hashtagStart = hashtagsMatches.start(1);
+            final int hashtagEnd = hashtagsMatches.end(1);
+            final String parsedHashtag = descriptionText.substring(hashtagStart, hashtagEnd);
+
+            spannableDescription.setSpan(new ClickableSpan() {
+                @Override
+                public void onClick(@NonNull final View view) {
+                    NavigationHelper.openSearch(context, streamingService.getServiceId(),
+                            parsedHashtag);
+                }
+            }, hashtagStart, hashtagEnd, 0);
+        }
+    }
+
+    /**
+     * Add click listeners which opens the popup player on timestamps in a plain text.
+     * <p>
+     * This method finds all timestamps in the {@link SpannableStringBuilder} of the description
+     * using a regular expression, adds for each a {@link ClickableSpan} which opens the popup
+     * player at the time indicated in the timestamps.
+     *
+     * @param context              the context to use
+     * @param spannableDescription the SpannableStringBuilder with the text of the
+     *                             content description
+     * @param contentUrl           the URL of the content
+     * @param streamingService     the {@link StreamingService} of the content
+     */
+    private static void addClickListenersOnTimestamps(final Context context,
+                                                      final SpannableStringBuilder
+                                                              spannableDescription,
+                                                      final String contentUrl,
+                                                      final StreamingService streamingService) {
+        final String descriptionText = spannableDescription.toString();
+        final Matcher timestampsMatches = TIMESTAMPS_PATTERN.matcher(descriptionText);
+
+        while (timestampsMatches.find()) {
+            final int timestampStart = timestampsMatches.start(2);
+            final int timestampEnd = timestampsMatches.end(3);
+            final String parsedTimestamp = descriptionText.substring(timestampStart, timestampEnd);
+            final String[] timestampParts = parsedTimestamp.split(":");
+            final int time;
+            if (timestampParts.length == 3) { // timestamp format: XX:XX:XX
+                time = Integer.parseInt(timestampParts[0]) * 3600 // hours
+                        + Integer.parseInt(timestampParts[1]) * 60 // minutes
+                        + Integer.parseInt(timestampParts[2]); // seconds
+                spannableDescription.setSpan(new ClickableSpan() {
+                    @Override
+                    public void onClick(@NonNull final View view) {
+                        playOnPopup(context, contentUrl, streamingService, time);
+                    }
+                }, timestampStart, timestampEnd, 0);
```

Comment:
Move this out of the `if else`, and add an `else continue` at the end to prevent (impossible) false matches